src/log/iobuf.rs

```diff
@@ -366,34 +402,13 @@ impl IOBufs {
         // signal that this IO buffer is uninitialized
         let max = std::usize::MAX as LogID;
         iobuf.set_log_offset(max);
-
-        // We spin here on trying to set the next buffer's disk offset.
-        // We need to spin here because we have the information about where
-        // its disk offset should be, and we choose to block here instead of when
-        // we write it in the future and it's not set up yet.
-        // Its offset is set to usize::MAX after it has been sucessfully written,
-        // so we must wait for this to be true before setting the new disk offset,
-        // otherwise we will alter the offset of pending writes.
-        let next_offset = log_offset + res_len as LogID;
-
-        let mut spins = 0;
-        loop {
-            spins += 1;
-            if spins > 1_000_000 {
-                // println!("{:?} have spun >1,000,000x in seal of buf {}", thread::current().name(), idx);
-                spins = 0;
-            }
-            if self.bufs[(idx + 1) % N_BUFS].cas_log_offset(max, next_offset).is_ok() {
-                // success, now we can stop stalling and do other work
-                break;
-            }
-        }
-
-        // NB allows new threads to start writing into this buffer
-        iobuf.set_header(0);
```

Comment:
this was a big performance problem because we were spinning in the call to `reserve` until the log offset was set for a buffer that we wanted to use in a reservation, but it wasn't being set until the last thread using the previous io buffer finished their operation and wrote the data to disk. this meant that as soon as an io buffer was sealed, there was essentially a global mutex being held, preventing new reservations from being made until possibly multiple threads finished their work in the last IO buffer.